src/dkg.ts

```diff
@@ -106,24 +112,57 @@ export class DkgClient {
     } as DkgRitual;
   }
 
-  // TODO: Without Validator public key in Coordinator, we cannot verify the
-  //    transcript. We need to add it to the Coordinator (nucypher-contracts #77).
-  // public async verifyRitual(ritualId: number): Promise<boolean> {
-  //   const ritual = await DkgCoordinatorAgent.getRitual(this.provider, ritualId);
-  //   const participants = await DkgCoordinatorAgent.getParticipants(
-  //     this.provider,
-  //     ritualId
-  //   );
-  //
-  //   const validatorMessages = participants.map((p) => {
-  //     const validatorAddress = EthereumAddress.fromString(p.provider);
-  //     const publicKey = FerveoPublicKey.fromBytes(fromHexString(p.???));
-  //     const validator = new Validator(validatorAddress, publicKey);
-  //     const transcript = Transcript.fromBytes(fromHexString(p.transcript));
-  //     return new ValidatorMessage(validator, transcript);
-  //   });
-  //   const aggregate = new AggregatedTranscript(validatorMessages);
-  //
-  //   return aggregate.verify(ritual.dkgSize, validatorMessages);
-  // }
+  public static async verifyRitual(
+    web3Provider: ethers.providers.Web3Provider,
+    ritualId: number
+  ): Promise<boolean> {
+    const ritual = await DkgCoordinatorAgent.getRitual(web3Provider, ritualId);
+    const participants = await DkgCoordinatorAgent.getParticipants(
+      web3Provider,
+      ritualId
+    );
+
+    // TODO: Does this check make sense here? Or do we delegate it to the Coordinator contract?
+    // for (const p of participants) {
+    //   // Not every participant has submitted a transcript
+    //   if (!p.aggregated) {
+    //     return false;
+    //   }
+    // }
```

Comment:
Perhaps you should just check the state of the ritual here - the expectation would that the ritual state is finalized. 

I believe that would be sufficient, and negate any specific checks for the participant.

Reply:
:heavy_check_mark: